### ISSUE TITLE
Contour doesn't work

### ISSUE THREAD

**Issue by swood** [projectcontour/contour#3775]
I'm trying to run the last version of Contour in Kubernetes with version 1.21.0.

I applied the yaml files and they created all resources. However, as far as I see the system doesn't work:

```
# kubectl  -n projectcontour get pod -o wide
NAME                       READY   STATUS             RESTARTS   AGE   IP             NODE             NOMINATED NODE   READINESS GATES
contour-5797bf86fd-rp5vz   0/1     CrashLoopBackOff   11         21m   172.100.4.40   ingress02   <none>           <none>
contour-5f79f7c5f4-9vn7m   0/1     CrashLoopBackOff   5          20m   172.100.3.46   ingress01   <none>           <none>
envoy-kmbs2                1/2     Running            0          26m   172.100.3.43   ingress01   <none>           <none>
envoy-wzld2                1/2     Running            0          26m   172.100.4.39   ingress02   <none>           <none>
```

No logs from contour pod. 
The only log I have is from envoy:
```
[2021-06-04 23:08:32.814][1][warning][config] [bazel-out/k8-opt/bin/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:101] StreamListeners gRPC config stream closed: 14, no healthy upstream
[2021-06-04 23:08:32.814][1][warning][config] [bazel-out/k8-opt/bin/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:63] Unable to establish new stream
```

But I missed the cause of the problem from it.

Could someone help me find the cause because I am trying to find it for 4 days and don't see it?

**Comment by youngnick** [comment]
Hi @swood, sorry to hear Contour isn't working for you, thanks for logging this issue. It looks like Contour is failing to start for some reason, you can see the pods are in CrashLoopBackoff. Could you check the logs for Contour and post them, or any weird logs you see?

Also, how did you build your Kubernetes cluster? Everything *should* work on Kubernetes 1.21, if there is something we've missed, we need to find it.

**Comment by skriss** [comment]
@swood do you still need help here? If so, please see Nick's previous comment requesting more information.

**Comment by stevesloka** [comment]
@swood just a friendly ping to see if we can help out here. Also can hop into the #contour channel on Kubernetes slack as well.

**Comment by swood** [comment]
@stevesloka hi, sorry, I didn't answer earlier.
Yes, I managed the problem, and the solution was tricky.
First, I had a non-standard API port.
Second, when I rolled back to the standard API port, I found in the Kubelet log that Envoy pod cannot assign pod-IP itself. That's how I found the second reason - I had two cni-configs at the same time. When I dropped unused cni-config - the problem has disappeared.

I think it might be helpful to add error logs for Contour pod as it is helpful to diagnose the core problem. In my case, I didn't see anything from that pod.
